### PR TITLE
Chore/update pr checklist

### PR DESCRIPTION
**Pull request by theStiggySpd**
Added a checklist for Pull Requests.
Please provide feedback to improve this checklist

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (Klaus-Keller) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:17]: I think this needs more explanation or I'm not even sure if it belongs to this checklist. 
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (Klaus-Keller) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:16]: This should be enforced by voters and code quality gates.
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (Klaus-Keller) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:14]:
```suggestion
- [ ] The code conforms to the general [development principles](https://github.com/SAP/open-ux-tools/blob/main/docs/Guidelines.md).
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (Klaus-Keller) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:18]: Couldn't find information about appropriate naming of branches. If we aim for branch names like:

```
fix|feat|docs|style|refactor|perf|test|chore|revert|WIP/<ISSUE_NUMBER>/<SHORT-DESCRIPTIO>
```
e.g.

```
fix/123/handle-exception
```

we should document it, perhaps in https://github.com/SAP/open-ux-tools/blob/main/docs/Guidelines.md
What do you think?

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (theStiggySpd) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Agreed. Removed it now https://github.com/SAP/open-ux-tools/pull/321/files#diff-45659dae3c8acfb7058b31c52c8600ca9c72a11a482372feea9ab85cb0f270fa

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (theStiggySpd) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: ok. I have made a replacement if it makes more sense to you?
`Already existing and new unit tests passes locally`

https://github.com/SAP/open-ux-tools/pull/321/files#diff-45659dae3c8acfb7058b31c52c8600ca9c72a11a482372feea9ab85cb0f270fa

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (theStiggySpd) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Done. https://github.com/SAP/open-ux-tools/pull/321/files#diff-45659dae3c8acfb7058b31c52c8600ca9c72a11a482372feea9ab85cb0f270fa

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (theStiggySpd) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: yes. this is now irrelevant as a checklist if we are providing link to dev guidelines. 
And this info belongs in the guidelines.

**Review by Klaus-Keller — APPROVED** [review]
The template for new pull requests looks reasonable. Approved from my side.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (ullasholla) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:14]: Checklist before asking for review/checklist before merging? I'm asking because it contains the item `...test pass locally`. For a PR to be merged in the tests have to pass on CI.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (ullasholla) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:17]: Just realized that we don't much about this in the dev guidelines. We probably need a section there

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (theStiggySpd) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:14]: hello @ullasholla, would you have better suggestion? goal is to get a good faith commitment from the pr author that they have made an attempt to ensure tests passes locally, not like using `--no-verify`. Thoughts?

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (theStiggySpd) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:17]: `we don't have much about this in the dev guidelines. `

Yes I noticed that too. May be after this pr goes through we can start adding some more guidelines there? This template should not change

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (ullasholla) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:14]: > goal is to get a good faith commitment from the pr author that they have made an attempt to ensure tests passes locally

In that case, this may not be required? PRs can't be merged in without tests passing and exceeding the coverage threshold. From my point of view, it's the PR author's responsibility to fix failing tests on CI, if any.

I see your point though of making them aware. I'm good either keeping it as it is or removing it entirely. Feel free to resolve the conversation

Comment (theStiggySpd) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:14]: ok point taken. Lets keep it for now and lets see some future PRs and we can adapt accordingly. thanks much

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (ullasholla) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:19]:
```suggestion
- [ ] Already existing and new unit tests pass locally
```

**Review by ullasholla — APPROVED** [review]
Thanks @theStiggySpd

**Comment by theStiggySpd** [comment]
> Thanks @theStiggySpd

thanks @ullasholla @Klaus-Keller 

Can i get one final approval from @SAP/ux-tools-admin so that we can merge?

**Review by tobiasqueck — APPROVED** [review]
Thank you.

**Pull request merged**
